packages/transformers/js/core/src/lib.rs

```diff
@@ -388,7 +384,12 @@ pub fn transform(
                   ),
                 );
 
-                module.fold_with(&mut passes)
+                let mut module = module.fold_with(&mut passes);
+                if config.source_type != SourceType::Script {
```

Comment:
Should this be done before `passes` so that the other optimisations (e.g. dead branch, paren remover, etc) are guaranteed to execute?

Reply:
Yes I think you're right; I've put it back in the same place surrounded with an `as_folder` which will wrap the visit-mut with a top-level Fold instance that just implements fold_module by calling into the visitor